src/components/Canvas/CanvasBuilder.jsx

```diff
@@ -1,17 +1,25 @@
 // @ts-check
 
 import React, { useEffect, useRef } from 'react'
+import { isFunction, isObject } from '@/utils/is'
 import styles from './Canvas.module.css'
 
 /**
- * @typedef  {Object} DrawFunctionConfig
- * @property {number} scale Calculated from device's pixel ratio
+ * @typedef  {Record<string, any>} DrawState
```

Comment:
`Record<>` is a Typescript thing. It works for me in VS Code because VS Code groks Typescript. But it is not strictly valid JSDoc syntax. The JSDoc way to type a plain generic object would be:

```suggestion
 * @typedef  {{ [key: string]: any }} DrawState
```

That's equivalent. If Vim loses its shit over `Record<string, any>` we can swap to `{ [key: string]: any }`